tests/php/src/Validation/URLValidationProviderTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace AmpProject\AmpWP\Tests\Validation;
+
+use AmpProject\AmpWP\Tests\Helpers\AssertContainsCompatibility;
+use AmpProject\AmpWP\Tests\Helpers\PrivateAccess;
+use AmpProject\AmpWP\Tests\Helpers\ValidationRequestMocking;
+use AmpProject\AmpWP\Validation\URLValidationProvider;
+use WP_UnitTestCase;
+
+/** @coversDefaultClass URLValidationProvider */
+final class URLValidationProviderTest extends WP_UnitTestCase {
+	use PrivateAccess, AssertContainsCompatibility;
+
+	/**
```

Comment:
Properties should be explicitly declared and not dynamically created.
```suggestion
	/**
	 * Validation provider instance to use.
	 *
	 * @var URLValidationProvider
	 */
	private $validation_provider;

	/**
```